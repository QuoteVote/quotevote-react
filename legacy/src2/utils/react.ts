import React, { 
  useState, 
  useEffect, 
  useRef, 
  ReactNode, 
  ChangeEvent, 
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
} from 'react';

export type { 
  ReactNode,
  ChangeEvent,
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
};

export { 
  React as default,
  useState,
  useEffect,
  useRef
};

// Add type declarations for common event handlers
export type InputChangeEvent = ChangeEvent<HTMLInputElement>;
export type TextAreaChangeEvent = ChangeEvent<HTMLTextAreaElement>;
export type ButtonClickEvent = MouseEvent<HTMLButtonElement>;
export type SetState<T> = Dispatch<SetStateAction<T>>;

// Add type declarations for common component props
export interface BaseProps {
  className?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}

export interface InputProps extends BaseProps {
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  disabled?: boolean;
} 