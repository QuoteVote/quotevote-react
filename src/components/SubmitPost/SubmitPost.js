import { useSelector } from 'react-redux'
import { useQuery } from '@apollo/react-hooks'
import React from 'react'
import { GROUPS_QUERY } from '../../graphql/query'
import SubmitPostSkeleton from './SubmitPostSkeleton'
import SubmitPostForm from './SubmitPostForm'

function SubmitPost(props) {
  const { setOpen } = props
  const user = useSelector((state) => state.user.data)
  const { loading, error, data } = useQuery(GROUPS_QUERY, {
    variables: { limit: 0 },
  })

  if (error) {
    return <div> Something went wrong!</div>
  }

  if (loading) {
    return <SubmitPostSkeleton />
  }

  const groupsOptions =
    (data &&
      data.groups.filter((group) => {
        const isUserAllowed = group.allowedUserIds.find((id) => id === user._id)
        return (
          group.privacy === 'public' ||
          (group.privacy === 'private' && isUserAllowed)
        )
      })) ||
    []

  return <SubmitPostForm options={groupsOptions} user={user} setOpen={setOpen}/>
}

export default SubmitPost
