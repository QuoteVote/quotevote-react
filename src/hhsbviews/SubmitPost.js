import React, { useState } from "react"
import { useHistory } from "react-router-dom"
import { useSelector } from 'react-redux'

import { 
  Grid,
  TextField, 
  Select,
  MenuItem,
  InputLabel,
  FormControl,
  Divider,
  InputBase,
  Typography,
  IconButton,
  Tooltip,
  Radio,
  RadioGroup,
} from "@material-ui/core"


import { makeStyles } from "@material-ui/core/styles"

import FormControlLabel from '@material-ui/core/FormControlLabel'
import FormLabel from '@material-ui/core/FormLabel'


// Materi Icons
import AddIcon from '@material-ui/icons/Add'
import RemoveIcon from '@material-ui/icons/Remove'
import AssignmentIcon from '@material-ui/icons/Assignment'


import GridItem from "material-ui/components/Grid/GridItem"
import Card from "material-ui/components/Card/Card"
import CardHeader from "material-ui/components/Card/CardHeader"
import CardBody from "material-ui/components/Card/CardBody"
import Button from "material-ui/components/CustomButtons/Button"
import LoadingSpinner from "hhsbComponents/LoadingSpinner.js"
import styles from "material-ui/assets/jss/material-dashboard-pro-react/views/sweetAlertStyle.js"

import GridContainer from "material-ui/components/Grid/GridContainer"

import SweetAlert from "react-bootstrap-sweetalert"
import * as copy from "clipboard-copy"


import { useQuery, useMutation } from '@apollo/react-hooks'
import { CREATE_DOMAIN, SUBMIT_TEXT } from 'graphql/mutations'
import { DOMAIN_QUERY } from 'graphql/query'

const useStyles = makeStyles(styles)

const inputStyles = {
  color: "#E91E63",
  fontSize: "25px",
  font: "League Spartan",
  fontWeight: "bold",
}

function SubmitPost() {
  const classes = useStyles()
  const [alert, setAlert] = React.useState(null)
  const [title, setTitle] = useState('[Enter Title]')
  const [text, setText] = useState('')
  const [domain, setDomain] = useState({domain: {
    title: 'Default',
  }})
  const [subScoreboardIsOpen, setCreateSubScoreboard] = useState(false)
  const [privacy, setPrivacy] = useState('private')

  let history = useHistory()

  const { user } = useSelector((state) => state.loginReducer)
  const { loading, error, data } = useQuery(DOMAIN_QUERY, {
    variables: { limit: 0 },
  })

  const [submitText, { data: submitData }] = useMutation(SUBMIT_TEXT)
  const [createDomain, { data: domainData}] = useMutation(CREATE_DOMAIN)

  const DOMAIN = process.env.REACT_APP_DOMAIN || 'localhost:3000'

  const handleSubmit =  async (event) => {
    event.preventDefault()
    let domainResult
    try {
      if (subScoreboardIsOpen) {
        domainResult= await createDomain({
          variables: {
            domain: {
              userId: user.creatorId,
              title: domain,
              url: `/${  domain.toLowerCase()}`,
              key: domain.toLowerCase(),
              privacy,
              description: `Description for: ${  domain  } group`,
            },
          },
        })
      }
      const domainId = domain._id ? domain._id : domainResult.data.createDomain._id
      const submitResult = await submitText({
        variables: {
          content: {
            title: title,
            text: text,
            creatorId: user.creatorId,
            domainId,
          },
        },
      })
      const { _id } = submitResult.data.addContent
      const domainKey = domain.key ? domain.key : domain.toLowerCase()
      successAlert(domainKey, _id)
    } catch (err) {
      errorAlert()
    }
  }

  const handleText = (event) => {
    setText(event.target.value)
  }

  const handleTitle = (event) => {
    setTitle(event.target.value)
  }

  const clearTitle = (event) => {
    if (title === '[Enter Title]') {
      setTitle('')
    }
  }

  const handleDomain = (event) => {
    setDomain(event.target.value)
  }

  const handleCreateSubScoreboard = () => {
    setDomain('')
    setCreateSubScoreboard(!subScoreboardIsOpen)
  }

  const handlePrivacy = (event) => {
    setPrivacy(event.target.value)
  }

  const successAlert = (domainKey, id) => {
    const shareableLink = `/boards/${domainKey}/content/${id}`

    setAlert(
      <SweetAlert
        success
        style={{ display: "block", top: "50%" }}
        title="You Created a Post!"
        onConfirm={() => history.push(shareableLink)}
        onCancel={() => hideAlert()}
        confirmBtnCssClass={`${classes.button} ${classes.success}`}
        confirmBtnText="Go to Post"
      >
        <Typography variant="caption">Share your text to your friends and family.</Typography>
        <Grid 
          container
          justify="space-around" 
          style={{ marginTop: 16 }}
          wrap="nowrap"
        >
          <GridItem style={{ overflow: 'auto' }}>
            <pre>{ DOMAIN + shareableLink }</pre>
          </GridItem>
          <GridItem style={{ flex: 1}}>
            <IconButton onClick={handleCopy(DOMAIN + shareableLink)}>
              <Tooltip title="Copy Link to Clip Board"><AssignmentIcon/></Tooltip>
            </IconButton>
          </GridItem>
        </Grid>
      </SweetAlert>
    )
  }
  
  const errorAlert = () => {
    setAlert(
      <SweetAlert
        error
        style={{ display: "block", top: "50%" }}
        title="Something went wrong!"
        onConfirm={() => hideAlert()}
        onCancel={() => hideAlert()}
        confirmBtnCssClass={`${classes.button  } ${  classes.danger}`}
        confirmBtnText="Ok"
      >
        We don't know what, yet let us know and we can find out
      </SweetAlert>
    )
  }
  const hideAlert = () => {
    setAlert(null)
  }

  const handleCopy = (shareableLink) => {
    copy(shareableLink)
  }

 
  if (loading) {
    return (
      <LoadingSpinner />
    )
  }
  
  if(error) {
    return (
      <SweetAlert
        error
        style={{ display: "block", top: "50%" }}
        title="Something went wrong!"
        onConfirm={() => hideAlert()}
        onCancel={() => hideAlert()}
        confirmBtnCssClass={`${classes.button  } ${  classes.danger}`}
        confirmBtnText="Ok"
      >
        We don't know what, yet let us know and we can find out
      </SweetAlert>
    )
  } 

  const userAllowedDomains = data.domains.filter((domain) => {
    const isUserAllowed = domain.allowedUserIds.find(
      (id) => id === user.creatorId
    )
    return (
      domain.privacy === "public" ||
        (domain.privacy === "private" &&
          isUserAllowed)
    )
  })

  return (
    <>
      {alert}
      <GridContainer spacing={1} direction="col">
        <GridItem xs={6}>
          <Card style={{ height: "800px" }}>
            <CardHeader style={{ zIndex: 0 }}>
              <div
                style={{
                  display: "flex",
                  direction: "row",
                  justifyContent: "space-between",
                  zIndex: 0,
                }}
              >
                <div
                  style={{
                    display: "flex",
                    direction: "row",
                    alignContent: "center",
                    justifyContent: "space-between",
                  }}
                >
                  <InputBase
                    style={inputStyles}
                    onClick={clearTitle}
                    onFocus={clearTitle}
                    onChange={handleTitle}
                    value={title}
                    name="title"
                    required
                  />
                </div>
                <div
                  style={{
                    display: "flex",
                    direction: "row",
                    justifyContent: "flex-end",
                    flexBasis: "100px",
                  }}
                >     
                </div>
              </div>
              <Divider />
            </CardHeader>
            <CardBody>
              <form
                onSubmit={handleSubmit}
              >
                <TextField
                  id="post"
                  label="Post"
                  placeholder="Input text to submit post"
                  multiline
                  fullWidth
                  name="text"
                  required
                  onChange={handleText}
                  value={text}
                  style={{
                    maxHeight: 600,
                    overflow: 'auto',
                  }}
                />
                <div style={{marginTop: 24, display: 'flex', justifyContent: 'space-between', alixgnContent: 'center'}}>
                  <div>
                    <FormControl required>
                      {
                        subScoreboardIsOpen && subScoreboardIsOpen                   ? 
                          <TextField
                            id="group"
                            label="Group"
                            placeholder="Create a new group"
                            name="group"
                            required
                            onChange={handleDomain}
                            value={domain.title}  
                            style={{ width: 280 }}
                          /> : 
                          <>
                            <InputLabel id="group-label" htmlFor="group">Group</InputLabel>
                            <Select
                              id="group"
                              value={domain.title}
                              placeholder={domain.title}
                              required
                              onChange={handleDomain}
                              style={{ width: 280 }}
                            >
                              <MenuItem value={domain.title}>
                                {domain.title}
                              </MenuItem>
                              {userAllowedDomains.map((publicDomain) => (
                                <MenuItem
                                  name={publicDomain.key}
                                  value={publicDomain}
                                  key={publicDomain._id}
                                >
                                  {publicDomain.title}
                                </MenuItem>))}
                            </Select>
                          </>
                      }
                
                    </FormControl>
                    <IconButton 
                      style={{ marginTop: 8, marginLeft: 4}}
                      onClick={handleCreateSubScoreboard}
                    >
                      { subScoreboardIsOpen && subScoreboardIsOpen ? 
                        <Tooltip title="Choose an existing group" style={{ fontSize: 18 }}><RemoveIcon/></Tooltip> :
                        <Tooltip title="Add a new group"><AddIcon/></Tooltip>
                      }
                    </IconButton>
                  </div>
                  <Button 
                    type="submit"
                    variant="contained"
                    size="large"
                    style={{
                      backgroundColor: 'rgb(233, 30, 99)',
                    }}
                  >
                Submit
                  </Button>
                </div>
                {
                  subScoreboardIsOpen &&
                <div style={{ paddingTop  : 16}}>
                  <FormControl required component="fieldset" >
                    <FormLabel component="legend">Choose Visibility</FormLabel>
                    <RadioGroup aria-label="privacy" name="privacy" value={privacy} onChange={handlePrivacy}>
                      <FormControlLabel value="private" control={<Radio />} label="Private" />
                      <FormControlLabel value="public" control={<Radio />} label="Pulbic" />
                    </RadioGroup>
                  </FormControl>
                </div>
                }
              </form>           
            </CardBody>
          </Card>
        </GridItem>
        <GridItem xs={6} style={{ paddingBottom: 0 }}>
          <Card>
            <CardHeader>
              <div
                style={{
                  display: "flex",
                  direction: "row",
                  justifyContent: "space-between",
                }}
              >
                <p
                  style={{
                    color: "#E91E63",
                    fontSize: "25px",
                    font: "League Spartan",
                    fontWeight: "bold",
                  }}
                >
                  Pop Prediction
                </p>
              </div>
            </CardHeader>
          </Card>
        </GridItem>
      </GridContainer>
    </>
  )
}

export default SubmitPost