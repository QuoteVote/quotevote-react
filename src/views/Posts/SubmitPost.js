/* eslint-disable no-underscore-dangle */
import React, { useState } from 'react'
import { useHistory } from 'react-router-dom'
import { useSelector, useDispatch } from 'react-redux'

import {
  Grid,
  TextField,
  Select,
  MenuItem,
  InputLabel,
  FormControl,
  Divider,
  InputBase,
  Typography,
  IconButton,
  Tooltip,
  Radio,
  RadioGroup,
} from '@material-ui/core'

import { makeStyles } from '@material-ui/core/styles'

import FormControlLabel from '@material-ui/core/FormControlLabel'
import FormLabel from '@material-ui/core/FormLabel'

// Material Icons
import AddIcon from '@material-ui/icons/Add'
import RemoveIcon from '@material-ui/icons/Remove'
import AssignmentIcon from '@material-ui/icons/Assignment'

import GridItem from 'mui-pro/Grid/GridItem'
import Card from 'mui-pro/Card/Card'
import CardHeader from 'mui-pro/Card/CardHeader'
import CardBody from 'mui-pro/Card/CardBody'
import Button from 'mui-pro/CustomButtons/Button'
import LoadingSpinner from 'components/LoadingSpinner'
import styles from 'assets/jss/material-dashboard-pro-react/views/sweetAlertStyle'

import GridContainer from 'mui-pro/Grid/GridContainer'

import SweetAlert from 'react-bootstrap-sweetalert'
import * as copy from 'clipboard-copy'
import { isEmpty } from 'lodash'

import { useQuery, useMutation } from '@apollo/react-hooks'
import { CREATE_GROUP, SUBMIT_POST } from 'graphql/mutations'
import { GROUPS_QUERY } from 'graphql/query'

import { SET_SELECTED_POST } from 'store/ui'
import { useForm } from 'react-hook-form'

const useStyles = makeStyles(styles)

const inputStyles = {
  color: '#E91E63',
  fontSize: '25px',
  font: 'League Spartan',
  fontWeight: 'bold',
  overflow: 'auto',
  width: '100%',
}

function SubmitPost() {
  const classes = useStyles()
  const [alert, setAlert] = useState(null)
  const [groupName, setGroupName] = useState('')
  const [groupId, setGroupId] = useState('')
  const [subScoreboardIsOpen, setCreateSubScoreboard] = useState(false)
  const [privacy, setPrivacy] = useState('private')
  const {
    register, handleSubmit, errors,
  } = useForm()

  const dispatch = useDispatch()

  const history = useHistory()

  const user = useSelector((state) => state.user.data)
  const { loading, error, data } = useQuery(GROUPS_QUERY, {
    variables: { limit: 0 },
  })
  const [submitPost] = useMutation(SUBMIT_POST)
  const [createGroup] = useMutation(CREATE_GROUP)

  const DOMAIN = process.env.REACT_APP_DOMAIN || 'localhost:3000'

  const onSubmit = async (values) => {
    const { title, text } = values
    try {
      let newGroup
      if (subScoreboardIsOpen) {
        newGroup = await createGroup({
          variables: {
            group: {
              creatorId: user._id,
              title: groupName,
              description: `Description for: ${groupName} group`,
              privacy,
            },
          },
        })
      }
      const postGroupId = newGroup ? newGroup.data.createGroup._id : groupId || userAllowedGroups[0]._id
      const submitResult = await submitPost({
        variables: {
          post: {
            userId: user._id,
            text,
            title,
            groupId: postGroupId,
          },
        },
      })
      const { _id, url } = submitResult.data.addPost
      dispatch(SET_SELECTED_POST(_id))
      successAlert(url, _id)
    } catch (err) {
      errorAlert(err)
    }
  }

  const handleCreateSubScoreboard = () => {
    setGroupName('')
    setCreateSubScoreboard(!subScoreboardIsOpen)
  }

  const handlePrivacy = (event) => {
    setPrivacy(event.target.value)
  }

  const successAlert = (shareableLink) => {
    // const shareableLink = `/hhsb/${domainKey}/content/${id}`

    setAlert(
      <SweetAlert
        success
        style={{ display: 'block', top: '50%' }}
        title="You Created a Post!"
        onConfirm={() => history.push(shareableLink)}
        onCancel={() => hideAlert()}
        confirmBtnCssClass={`${classes.button} ${classes.success}`}
        confirmBtnText="Go to Post"
      >
        <Typography variant="caption">
          Share your text to your friends and family.
        </Typography>
        <Grid
          container
          justify="space-around"
          style={{ marginTop: 16 }}
          wrap="nowrap"
        >
          <GridItem style={{ overflow: 'auto' }}>
            <pre>{DOMAIN + shareableLink}</pre>
          </GridItem>
          <GridItem style={{ flex: 1 }}>
            <IconButton onClick={handleCopy(DOMAIN + shareableLink)}>
              <Tooltip title="Copy Link to Clip Board">
                <AssignmentIcon />
              </Tooltip>
            </IconButton>
          </GridItem>
        </Grid>
      </SweetAlert>
    )
  }

  const errorAlert = (err) => {
    setAlert(
      <SweetAlert
        error
        style={{ display: 'block', top: '50%' }}
        title="Something went wrong!"
        onConfirm={() => hideAlert()}
        onCancel={() => hideAlert()}
        confirmBtnCssClass={`${classes.button} ${classes.danger}`}
        confirmBtnText="Ok"
      >
        {/* We don't know what, yet let us know and we can find out */}
        Error:
        {' '}
        {err}
      </SweetAlert>
    )
  }
  const hideAlert = () => {
    setAlert(null)
  }

  const handleCopy = (shareableLink) => {
    copy(shareableLink)
  }

  if (loading) {
    return <LoadingSpinner />
  }

  if (error) {
    return (
      <SweetAlert
        error
        style={{ display: 'block', top: '50%' }}
        title="Something went wrong!"
        onConfirm={() => hideAlert()}
        onCancel={() => hideAlert()}
        confirmBtnCssClass={`${classes.button} ${classes.danger}`}
        confirmBtnText="Ok"
      >
        We don&apos;t know what, yet let us know and we can find out
      </SweetAlert>
    )
  }

  const userAllowedGroups =
    (data &&
      data.groups.filter((group) => {
        const isUserAllowed = group.allowedUserIds.find((id) => id === user._id)
        return (
          group.privacy === 'public' ||
          (group.privacy === 'private' && isUserAllowed)
        )
      })) ||
    []

  const handleSetGroupId = (event) => {
    if (event.target.value) {
      setGroupId(event.target.value)
    }
  }
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {alert}
      <GridContainer>
        <GridItem xs={12} xl={6}>
          <Card style={{ height: '600px' }}>
            <CardBody>
              <InputBase
                fullWidth
                data-testid="title-input"
                id="title"
                style={inputStyles}
                placeholder="[Enter Title]"
                name="title"
                inputRef={register({
                  required: 'Title is required',
                })}
                required
                error={errors.title}
                helperText={errors.title && errors.title.message}
              />
              <Divider />
              <TextField
                id="text"
                label="Post"
                placeholder="Input text to submit post"
                multiline
                fullWidth
                rows={3}
                rowsMax={20}
                name="text"
                style={{
                  maxHeight: 600,
                  overflow: 'auto',
                }}
                inputRef={register({
                  required: 'Post is required',
                })}
                required
                error={errors.text}
              />
              <div
                style={{
                  marginTop: 24,
                  display: 'flex',
                  justifyContent: 'space-between',
                  alignContent: 'center',
                  flexWrap: 'wrap',
                }}
              >
                <div>
                  <FormControl required>
                    {subScoreboardIsOpen && subScoreboardIsOpen ? (
                      <TextField
                        id="group"
                        name="group"
                        label="Group"
                        placeholder="Create a new group"
                        value={groupName}
                        style={{ width: 280 }}
                        inputRef={register({
                          required: 'Group is required',
                        })}
                        error={errors.group}
                      />
                    ) : (
                      <>
                        <InputLabel id="group-label" htmlFor="group">
                          Group
                        </InputLabel>
                        <Select
                          data-testid="group"
                          name="group"
                          id="group"
                          value={groupId || userAllowedGroups[0]._id}
                          placeholder={groupName}
                          required
                          onChange={handleSetGroupId}
                          style={{ width: 280 }}
                        >
                          <MenuItem value={groupName}>{groupName}</MenuItem>
                          {!isEmpty(userAllowedGroups) &&
                          userAllowedGroups.map((publicGroup) => (
                            <MenuItem
                              value={publicGroup._id}
                              key={publicGroup._id}
                            >
                              {publicGroup.title}
                            </MenuItem>
                          ))}
                        </Select>
                      </>
                    )}
                  </FormControl>
                  <IconButton
                    style={{ marginTop: 8, marginLeft: 4 }}
                    onClick={handleCreateSubScoreboard}
                  >
                    {subScoreboardIsOpen && subScoreboardIsOpen ? (
                      <Tooltip
                        title="Choose an existing group"
                        style={{ fontSize: 18 }}
                      >
                        <RemoveIcon />
                      </Tooltip>
                    ) : (
                      <Tooltip title="Add a new group">
                        <AddIcon />
                      </Tooltip>
                    )}
                  </IconButton>
                </div>
                <Button
                  id="submit-button"
                  type="submit"
                  variant="contained"
                  size="large"
                  style={{
                    backgroundColor: 'rgb(233, 30, 99)',
                    marginTop: '8px',
                  }}
                >
                  Submit
                </Button>
              </div>
              {subScoreboardIsOpen && (
                <div style={{ paddingTop: 16 }}>
                  <FormControl required component="fieldset">
                    <FormLabel component="legend">
                      Choose Visibility
                    </FormLabel>
                    <RadioGroup
                      aria-label="privacy"
                      name="privacy"
                      value={privacy}
                      onChange={handlePrivacy}
                    >
                      <FormControlLabel
                        value="private"
                        control={<Radio />}
                        label="Private"
                      />
                      <FormControlLabel
                        value="public"
                        control={<Radio />}
                        label="Public"
                      />
                    </RadioGroup>
                  </FormControl>
                </div>
              )}
            </CardBody>
          </Card>
        </GridItem>
        <GridItem xs={12} xl={6} style={{ paddingBottom: 0 }}>
          <Card>
            <CardHeader>
              <div
                style={{
                  display: 'flex',
                  direction: 'row',
                  justifyContent: 'space-between',
                }}
              >
                <p
                  style={{
                    color: '#E91E63',
                    fontSize: '25px',
                    font: 'League Spartan',
                    fontWeight: 'bold',
                  }}
                >
                  Pop Prediction
                </p>
              </div>
            </CardHeader>
          </Card>
        </GridItem>
      </GridContainer>
    </form>
  )
}

export default SubmitPost
