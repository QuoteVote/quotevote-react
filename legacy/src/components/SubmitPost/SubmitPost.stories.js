// Important stuff, must always be imported on a storybook file
import React from 'react'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withA11y } from '@storybook/addon-a11y'

// Apollo Imports not needed for story, just for calling data
// If you want to apply theme
// The component
import SubmitPost from './SubmitPost'

// Story config
export default {
  title: 'Submit Post',
  component: SubmitPost,
  decorators: [withKnobs, withA11y],
}

export const SubmitPostWithOptions = () => <SubmitPost options={[]} user={{ name: 'John Doe', avatar: '' }} />

SubmitPostWithOptions.story = {
  parameters: {
    jest: ['SubmitPost.test.js'],
  },
}
