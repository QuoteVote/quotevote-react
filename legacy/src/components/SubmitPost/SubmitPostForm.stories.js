// Important stuff, must always be imported on a storybook file
import React from 'react'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withA11y } from '@storybook/addon-a11y'

// Apollo Imports not needed for story, just for calling data
// If you want to apply theme
// The component
import SubmitPostForm from './SubmitPostForm'

// Story config
export default {
  title: 'Submit Post',
  component: SubmitPostForm,
  decorators: [withKnobs, withA11y],
}

export const Form = () => <SubmitPostForm />

Form.story = {
  parameters: {
    jest: ['SubmitPostForm.test.js'],
  },
}
