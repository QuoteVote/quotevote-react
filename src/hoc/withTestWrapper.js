import React from 'react'
import { PersistGate } from 'redux-persist/integration/react'
import { ApolloProvider } from '@apollo/react-hooks'
import { Provider } from 'react-redux'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import store, { persistor } from '../store/store'
import client from '../config/apollo'
import customTheme from '../theme'
const hist = createBrowserHistory()
const theme = createTheme(customTheme)

const withTestWrapper = (Component) => (props) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <ThemeProvider theme={theme}>
          <Router history={hist}>
            <Component {...props} />
          </Router>
        </ThemeProvider>
      </PersistGate>
    </Provider>
  </ApolloProvider>
)
export default withTestWrapper
