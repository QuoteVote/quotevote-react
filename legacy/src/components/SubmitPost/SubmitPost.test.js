import React from 'react'
import { render } from '@testing-library/react'

// Component being tested
import SubmitPost from './SubmitPost'
import withTestWrapper from '../../hoc/withTestWrapper'

const SubmitPostWrapper = withTestWrapper(SubmitPost)

describe('SubmitPost test -', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SubmitPostWrapper />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
