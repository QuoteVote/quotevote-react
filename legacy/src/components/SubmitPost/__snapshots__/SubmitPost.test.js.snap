// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitPost test - renders correctly 1`] = `
<div
  class="makeStyles-card-1"
>
  <div
    class="makeStyles-cardBody-18"
  >
    <div
      class="MuiTypography-root MuiTypography-h2"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-h4"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-h4"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-caption"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="width: 100px; height: 35px;"
      />
    </div>
  </div>
</div>
`;
